src/Diffusion.jl

```diff
@@ -61,6 +61,7 @@ end
 
 Base.@ccallable function julia_diffusion(nx::Csize_t, ny::Csize_t, nt::Csize_t, lx::Cdouble, ly::Cdouble, lam::Cdouble, cp_min::Cdouble, amp_T::Cdouble, Cp_c::CuPtr{Cdouble}, comm_c::MPI.MPI_Comm)::Cint
     try
+        MPI.run_init_hooks()
```

Comment:
@simonbyrne , can you confirm that this is the right place to call it? Is there also some `MPI.run_finalize_hooks()` to add?

Reply:
You want to only run those once. So they should be called from C.

Reply:
@vchuravy , this means that a C interface like the following needs to be created in the Julia part for this?
```julia
Base.@ccallable function julia_run_mpiinit_hooks()::Cint
    try
        MPI.run_init_hooks()
    catch
        Base.invokelatest(Base.display_error, Base.catch_stack())
        return 1
    end
    return 0
end
```

Reply:
Yes that should work.

There aren't currently any finalize hooks.

Reply:
You can also forgo the c-interface and use `jl_apply`

Reply:
@vchuravy : `jl_apply` takes a string (I could not find a documentation non that)? So would it just mean to add the following line in main.c after `MPI_Init()`?
```
jl_apply("MPI.run_init_hooks()")
```

Reply:
Ah sorry the high-level API for that is `jl_call`, `jl_apply` is the internal variant. https://docs.julialang.org/en/v1/manual/embedding/#Calling-Julia-Functions

So the easiest here is just to call `jl_eval_string`.